operator-common/src/main/java/io/strimzi/operator/PlatformFeaturesAvailability.java

```diff
@@ -56,13 +69,57 @@
             return checkApiAvailability(vertx, httpClient, client.getMasterUrl().toString(), "image.openshift.io", "v1");
         }).compose(supported -> {
             pfa.setImages(supported);
+            return highestEventVersion(vertx, httpClient, client.getMasterUrl().toString());
+        }).compose(eventApiVersion -> {
+            pfa.setHighestEventApiVersion(eventApiVersion);
             return Future.succeededFuture(pfa);
         })
         .onComplete(pfaPromise);
 
         return pfaPromise.future();
     }
 
+    /**
+     * Brief rundown - the various APIs and K8s versions
+     *   events.k8s.io/v1      - introduced in Kubernetes 1.19
+     *   events.k8s.io/v1beta1 - introduced in Kubernetes 1.8, deprecated in 1.19, will be removed in 1.25
+     *   core events           - Always around, but has significant performance issues, so a new event API was introduced.
```

Comment:
Why do we need the core events? It is parallel to the `events.k8s.io` so I think we should use one or another. My undertsanding is that `events.k8s.io` is _ewer and better_ - so why not just use that? What are the situation when you expect us to use core events?

Reply:
When someone is using Kubernetes < 1.19 and has disabled either the beta events.k8s.io API, or all beta APIs using `--feature-gates AllBeta=true` as an argument to the api-server.

Reply:
Well, did you checked if Strimzi really works in that case? We do not do it for anything else and I doubt this would be the only problem then. So it is maybe not worth the effort? (Plus if you see it like this, users can disable any API IIRC)

Reply:
I have only checked it worked on 1.16.0 with the beta events API disabled. I'll fire up Minikube now with all beta APIs disabled, see what happens :D

Reply:
[So I gave it a try](https://gist.github.com/LiamClarkeNZ/9c7fda2fdcf48b7533f441ca8c2c0ee9), and Strimzi does hit an issue with one beta API, but K8s itself hit far more issues - so I'd be happy to a) document that K8s events require events.k8s.io/v1beta1 or events.k8s.io/v1 and b) log at WARN level when neither is available.

Reply:
> a) document that K8s events require events.k8s.io/v1beta1 or events.k8s.io/v1 and

TBH, we do not document this for any other API. We also haven't run into anyone with disabled beta APIs so far.

Reply:
Cool, 'll leave it as a log.warn() then.